plugins/languages/treesitter/treesitter.nix

```diff
@@ -411,9 +416,10 @@ helpers.neovim-plugin.mkNeovimPlugin config {
 
     extraFiles = mkIf cfg.nixvimInjections { "queries/nix/injections.scm".source = ./injections.scm; };
 
-    extraPlugins = mkIf (cfg.package != null) [
+    extraPlugins = mkIf (cfg.package != null || cfg.iconsPackage != null) [
```

Comment:
This feels a bit messy. Could we use mkMerge here, or perhaps just concat optional lists?

Thinking about it perhaps we don't need to check `!= null` anyway, because our `extraPlugins` option already takes care of that for us... 🤔

Reply:
Yeah, I think I started writing lib.optionals and couldn't remember if there was a reason for the null check.. i think it was before you added the null check into the mkNeovimPlugin...

Reply:
So, iconsPackage accepts a null if someone doesn't wish to install the package. But, extraPlugins doesn't so we need to ignore setting it when it's null.